### ISSUE TITLE
cannot delete list directly `del [x]`

### ISSUE THREAD

**Issue by key262yek** [RustPython/RustPython#3878]
## Feature

In CPython, they can delete a list in following manner

```python3
>>> x = 1
>>> del [x]
```

But, in RustPython we cannot.
```python3
>>> del [x]
SyntaxError: cannot delete list at line 1 column 1
del [x]
^
```

Related test is in `Lib/test/test_syntax.py`.
```python3
def test_assign_del(self):
        # ...
        self._check_error("del [*x,]", "cannot delete starred")    # Not "cannot delete list"
```

## Python Documentation

Cannot find documentation exactly related this difference.

**Comment by key262yek** [comment]
related to #3879

**Comment by key262yek** [comment]
I add match to List in compile_delete() in file compile/src/compile.rs

```rust
        ast::ExprKind::Tuple { elts, .. } | ast::ExprKind::List { elts, .. } => {
                for element in elts {
                    self.compile_delete(element)?;
                }
```

pr #3879 includes the code changes.

**Comment by key262yek** [comment]
It is resolved by pr #3879
